homeassistant/components/philips_js/media_player.py

```diff
@@ -81,70 +85,69 @@ def _inverted(data):
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities,
+):
+    """Set up the configuration entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        [
+            PhilipsTVMediaPlayer(
+                coordinator,
+                config_entry.data[CONF_SYSTEM],
+                config_entry.unique_id or config_entry.entry_id,
+            )
+        ]
+    )
 
 
-class PhilipsTVMediaPlayer(MediaPlayerEntity):
+class PhilipsTVMediaPlayer(CoordinatorEntity, MediaPlayerEntity):
     """Representation of a Philips TV exposing the JointSpace API."""
 
-    def __init__(self, tv: PhilipsTV, name: str, on_script: Script):
+    def __init__(
+        self,
+        coordinator: PhilipsTVDataUpdateCoordinator,
+        system: Dict[str, Any],
+        unique_id: str,
+    ):
         """Initialize the Philips TV."""
-        self._tv = tv
-        self._name = name
+        self._tv = coordinator.api
+        self._coordinator = coordinator
         self._sources = {}
         self._channels = {}
-        self._on_script = on_script
         self._supports = SUPPORT_PHILIPS_JS
-        if self._on_script:
-            self._supports |= SUPPORT_TURN_ON
-        self._update_task = None
+        self._system = system
+        self._unique_id = unique_id
+        super().__init__(coordinator)
+        self._update_from_coordinator()
 
-    def _update_soon(self, delay):
+    def _update_soon(self):
         """Reschedule update task."""
-        if self._update_task:
-            self._update_task()
-            self._update_task = None
-
-        self.schedule_update_ha_state(force_refresh=False)
-
-        def update_forced(event_time):
-            self.schedule_update_ha_state(force_refresh=True)
-
-        def update_and_restart(event_time):
-            update_forced(event_time)
-            self._update_task = track_time_interval(
-                self.hass, update_forced, timedelta(seconds=DEFAULT_SCAN_INTERVAL)
-            )
-
-        call_later(self.hass, delay, update_and_restart)
-
-    async def async_added_to_hass(self):
-        """Start running updates once we are added to hass."""
-        await self.hass.async_add_executor_job(self._update_soon, 0)
+        self.hass.add_job(self.coordinator.async_request_refresh)
 
     @property
     def name(self):
         """Return the device name."""
-        return self._name
-
-    @property
-    def should_poll(self):
-        """Device should be polled."""
-        return False
+        return self._system["name"]
 
     @property
     def supported_features(self):
         """Flag media player features that are supported."""
-        return self._supports
+        supports = self._supports
+        if self._coordinator.turn_on:
```

Comment:
This one was a bit unexpected, but it seems better then selecting script entity so LGTM